andela/magma-backend
# #167749952 Implement user signin endpoint

#### What does this PR do?

This PR implements the user signin API endpoint

#### Description of Tasks to be completed

- setup test suite for user signin endpoint
- create a function to generate token
- create a function to compare password with hash password
- create a signin function in the UserService class to get user information by email
- implements signin endpoint in the UserController class
- add the route
- test with postman
- ensure all test cases pass

#### How should this be manually tested?

- Clone the Repo, change into the directory and run `npm install`
- Run `npm test` to see all test cases
- Run `npm dev` to start the server
- Make a `POST` request to `http://localhost:{PORT}/api/v1/user/signin` on Postman using valid credentials

#### Relevant pivotal tracker story

[167749952](https://www.pivotaltracker.com/story/show/167749952)
